testing/test_crossbrowsertesting.py

```diff
@@ -33,12 +34,37 @@ def failure(testdir, testfile, httpserver_base_url):
                    '--driver', 'CrossBrowserTesting')
 
 
-def test_missing_username(failure):
-    out = failure()
-    assert 'UsageError: CrossBrowserTesting username must be set' in out
+def test_missing_username(failure, monkeypatch, tmpdir):
+    monkeypatch.setattr(os.path, 'expanduser', lambda p: str(tmpdir))
+    assert 'UsageError: CrossBrowserTesting username must be set' in failure()
 
 
-def test_missing_api_key(failure, monkeypatch):
+def test_missing_access_key_env(failure, monkeypatch, tmpdir):
+    monkeypatch.setattr(os.path, 'expanduser', lambda p: str(tmpdir))
     monkeypatch.setenv('CROSSBROWSERTESTING_USERNAME', 'foo')
+    assert 'UsageError: CrossBrowserTesting auth key must be set' in failure()
+
+
+def test_missing_access_key_file(failure, monkeypatch, tmpdir):
+    credentials = tmpdir.join('.crossbrowsertesting')
+    credentials.write('[credentials]\nusername=foo')
+    monkeypatch.setattr(os.path, 'expanduser', lambda p: str(credentials))
+    assert 'UsageError: CrossBrowserTesting auth key must be set' in failure()
+
+
+def test_invalid_credentials_env(failure, monkeypatch, tmpdir):
+    monkeypatch.setattr(os.path, 'expanduser', lambda p: str(tmpdir))
```

Comment:
If you refactor `credentials()` so it can be reused, I suggest then refactoring the tests as well so you can use the same set of tests related to credentials in all the drivers. 

I usually do this by having a base class with test methods, and then subclass that base class in each module configuring it (perhaps using class variables) with the specifics for that driver.